data/core/macros/unit-utils.cfg

```diff
@@ -253,3 +253,766 @@
         transform_to={TYPE}
     [/transform_unit]
 #enddef
+
```

Comment:
Not sure if it's the best place, but fine for now.
All [event]s need IDs. That way, if the macro is inlcuded in an [era] and a [scenario] (or even [campaign]), the event will only be included once.

Reply:
you are pull FORMATION ID GREATER_THAN COUNT in unit-utils.cfg, see what it is an ability include in formation you must be pull that in abilities.cfg instead. For macros abilities event, maybe you doing best to create a abilities-event.cfg.